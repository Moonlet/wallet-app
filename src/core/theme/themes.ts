export enum Theme {
    dark = 'dark',
    light = 'light'
}
