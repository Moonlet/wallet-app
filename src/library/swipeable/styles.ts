import { StyleSheet } from 'react-native';
import { ITheme } from '../../core/theme/itheme';
import { normalize } from '../../styles/dimensions';

export default (theme: ITheme) =>
    StyleSheet.create({
        container: {
            borderWidth: 1,
            borderColor: theme.colors.accent,
            flexDirection: 'row',
            width: '100%',
            height: normalize(40)
        },

        tabButton: {
            borderWidth: 1,
            borderColor: theme.colors.accent,
            flexBasis: 0,
            flex: 1,
            alignItems: 'center',
            justifyContent: 'center'
        },
        tabButtonText: {},
        tabButtonSelected: {
            backgroundColor: theme.colors.accent
        },
        tabButtonTextSelected: {
            color: theme.colors.appBackground
        }
    });
