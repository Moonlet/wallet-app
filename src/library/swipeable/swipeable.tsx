import React from 'react';

import stylesProvider from './styles';
import { withTheme } from '../../core/theme/with-theme';
import RNGHSwipable from 'react-native-gesture-handler/Swipeable';

export interface ISwipableProps {
    children?: any;
    renderLeftActions?: any;
    styles: ReturnType<typeof stylesProvider>;
}

/**
 * // TODO this is not used at the moment, maybe remove it if not needed for chrom ext
 * @param props
 */
export const SwipeableComponent = (props: ISwipableProps) => (
    <RNGHSwipable renderLeftActions={props.renderLeftActions}>{props.children}</RNGHSwipable>
);

export const Swipeable = withTheme(stylesProvider)(SwipeableComponent);
