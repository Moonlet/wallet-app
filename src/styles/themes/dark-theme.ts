import { ITheme } from '../../core/theme/itheme';
import { COLORS } from '../colors';

export const darkTheme: ITheme = {
    dark: true,
    shadowGradient: ['#00000000', '#000000CC'],

    fontSize: {
        small: 12,
        regular: 16,
        large: 34
    },

    colors: {
        primary: COLORS.GRAY,
        accent: COLORS.AQUA,
        text: COLORS.WHITE,
        textDarker: COLORS.LIGHT_GRAY,
        positive: COLORS.HOT_GREEN,
        negative: COLORS.HOT_RED,

        cardBackground: COLORS.DARK_GRAY,
        appBackground: COLORS.COD_GRAY,
        settingsDivider: COLORS.BLACK
    }
};
