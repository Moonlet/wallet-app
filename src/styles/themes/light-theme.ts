import { ITheme } from '../../core/theme/itheme';
import { darkTheme } from './dark-theme';

export const lightTheme: ITheme = darkTheme;
