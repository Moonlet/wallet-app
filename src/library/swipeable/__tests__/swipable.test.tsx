import React from 'react';
import { Swipeable, ISwipableProps } from '../swipeable';
import { darkTheme } from '../../../styles/themes/dark-theme';
import styleProvider from '../styles';

import { shallow } from 'enzyme';

const props: ISwipableProps = {
    styles: styleProvider(darkTheme),
    children: null,
    renderLeftActions: () => 'left'
};

describe('library checkbox', () => {
    it('renders correctly', () => {
        const wrapper = shallow(<Swipeable {...props} />);
        expect(wrapper.debug()).toMatchSnapshot();
    });
});
