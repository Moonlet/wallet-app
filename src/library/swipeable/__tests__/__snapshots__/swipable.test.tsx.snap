// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`library checkbox renders correctly 1`] = `"<SwipeableComponent styles={{...}} renderLeftActions={[Function: renderLeftActions]} theme={{...}} />"`;
